elements/lisk-codec/src/varint.ts

```diff
@@ -124,8 +124,11 @@ const validateVarintSize = (result: bigint, size: number) => {
 	}
 
 	const min = BigInt(1) << BigInt(7 * (size - 1));
+	if (result < min) {
+		throw new Error('invalid varint bytes. vartint must be in shortest form.');
+	}
 	const max = BigInt(1) << BigInt(7 * size);
-	if (result < min || result >= max) {
+	if (result >= max) {
```

Comment:
can this actually happen?

Reply:
mmm that's a good point, i think longer form is always caught on the min